inst/tutorials/practical0-1/practical0-1.Rmd

```diff
@@ -0,0 +1,444 @@
+cause hours of agony to some students every year on this course. As a result,
+we insist on two things in your submitted practicals:
+
+1. You use variable names inside functions that you don't use outside in your
+   scripts wherever possible. Sometimes there really only is one sensible name
+   for a variable (maybe `timestep` when describing the increments in your
+   simulation model), and you should be very careful when that happens, but
+   normally there are at least two ways of describing the same idea - often one
+   more general in a function (`birth.rate`) which can be used in multiple
+   contexts, and then specific in your main script each time you use it
+   (`human.annual.birth.rate`, for instance).
+2. You write checks for your functions to make sure they aren't accidentally
+   missing any variables or arguments. There are two tools for this,
+   `findGlobals()` and `checkUsage()` in the `codetools` library.
+
+Try adding the following lines to the end of *Exercise 3*:
+
+```{r, eval=FALSE}
+library(codetools)
+checkUsage(misspelled_add)
+```
+
+You'll see that it reports on the problem that you have probably already
+identified. It can also identify other problems, and we recommend it to you.
+However, it doesn't return anything you can use in your code to allow us to
+automatically reject bad functions, so there is a second solution we'll
+use in our code, and we'll show you how to use it in a later exercise, but
+for now just try using this line instead of `checkUsage()` above (the
+`library()` call is the same):
+
+```{r, eval=FALSE}
+findGlobals(misspelled_add, merge = FALSE)
+```
+
+You'll see that it identifies the use of the global variable `result` too
+(if less elegantly). We'll show you how to use this to automatically detect
+problems with variable misnaming in a later exercise, but for now just remember
+that you need to watch out for misspelling very carefully.
+
+*If you're interested, the `findGlobals()` output also tells you why this
+problem exists in R. You'll see that the function `misspelled_add()` needs
+to use three other *global* variables - the functions `{`, `+` and `<-` - to get
+the function to work. As a result of the way R works, functions have to be able
+to see what is going on outside to be able to work at all. Most languages handle
+this more elegantly.*
+
+## Loops
+
+Loops allow you to do the same thing multiple times easily. You *loop* over the
+same bit of code again and again instead of having to copy and paste the code
+multiple times into your script. We'll talk more about different kinds of loops
+in a later practical, but for now we'll focus on the simplest - something that
+tells you to do something a fixed number of times that you know in advance.
+This is called a `for` loop.
+
+A `for` loop always has the same style:
+
+```{r, eval=FALSE}
+for (one in many) {
+  # Do something
+    ... exciting code maybe using the variable
+        one (but it doesn't have to) ...
```

Comment:
I'd comment this out. If the students aren't paying attention, they might think it's ok to comment without a hash. But also, the crappy code highlighting picks up bits of it :(